.github/workflows/publish-docs.yaml

```diff
@@ -0,0 +1,25 @@
+name: Publish docs via GitHub Pages
```

Comment:
This entire file was renamed from `main.yml` and with loads of quotes etc making it hard to read YAML.

I updated the mixed use of "build" "deploy" "publish" to just "publish".